src/IR/LambdaLift.hs

```diff
@@ -209,20 +239,31 @@ liftLambdas (I.Match s arms t) = do
 liftLambdas lit@I.Lit{}  = return lit
 liftLambdas dat@I.Data{} = return dat
 
+-- | Entry point for traversing let bindings.
+liftLambdasInLet :: (I.Binder, I.Expr Poly.Type) -> LiftFn (I.Expr Poly.Type)
+liftLambdasInLet (b, expr) = do
+  let bindingName = maybe "wildcard" ident b
```

Comment:
Don't descend with `"wildcard"` here; instead, just descend with `Nothing`.

Otherwise, you'll have a lot of `wildcard`s mixed into your mangled identifiers, because things like `a ; b` are desugared to `let _ = a ; b`